src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs

```diff
@@ -26,6 +26,11 @@ public static void CallEntrypoint(JSMarshalerArgument* arguments_buffer)
             ref JSMarshalerArgument arg_2 = ref arguments_buffer[3]; // initialized and set by caller
             try
             {
+#if FEATURE_WASM_THREADS
+                // when we arrive here, we could assume that all proxies are owned by calling thread
+                JSProxyContext.CapturedInstance = JSProxyContext.AssertCurrentContext();
```

Comment:
It would be ideal if this didn't need to be ifdef'd/conditional so it's easier to just write the correct thing everywhere, but that's not a blocker

Reply:
this is correct for ST, but it's unnecessary instructions to  download and execute